I've revised the original example for calling `queryEvents` with the Sui TypeScript SDK. (#16082)
MystenLabs/sui: docs/content/guides/developer/sui-101/using-events.mdx

### TypeScript

To create the event subscription, you can use a basic Node.js app. You need the Sui TypeScript SDK, so install the module using `npm install @mysten/sui.js` at the root of your project. In your TypeScript code, Use queryEvents within the executeMoveCall function below to query the events information for a Transaction.Below is an example of how to obtain the events information for the current transaction when calling a contract.

Example code：

```ts
import {
    useSignAndExecuteTransactionBlock,
    useSuiClient,
  } from "@mysten/dapp-kit";
import { TransactionBlock } from "@mysten/sui.js/transactions";
import { Button, Flex, Heading } from "@radix-ui/themes";

> import { SICBOGAME_PACKAGE_ID } from "./constants";  

Reviewer: ```suggestion

```

  
export function Creategame({ 
    onCreated,
}:{
    onCreated: (id: string) => void;
 }) {
    const { mutate: signAndExecute } = useSignAndExecuteTransactionBlock();
    const client = useSuiClient();

    const executeMoveCall = (method: "small" | "large") => {
        const txb = new TransactionBlock();

        if (method === "small") {
        txb.moveCall({
            arguments: [txb.pure.u64(0)],

>             target: `${SICBOGAME_PACKAGE_ID}::sicbogame::create_game`,

Reviewer: ```suggestion
            target: `<PACKAGE-ID>::<MODULE>::create_game`,
```

        });
        } else {
        txb.moveCall({
            arguments: [txb.pure.u64(1)],

>             target: `${SICBOGAME_PACKAGE_ID}::sicbogame::create_game`,

Reviewer: ```suggestion
            target: `<PACKAGE-ID>::<MODULE>::create_game`,
```

        {
            onSuccess: (tx) => {
                client.
                    queryEvents({
                        query:{Transaction:tx.digest}
                    }).then(PaginatedEvents=> {
                        const events = JSON.stringify(PaginatedEvents.data[0].parsedJson);
                    
                        if (events.length > 0) {
                          onCreated(events)
                        } else {
                          onCreated('No events found for the given criteria.');
                        }
                      })
                      .catch(error => { 
                        console.error('Error querying events:', error);
                      });
            },
        },
        );
    };
    return (
        <>
        <Heading size="3">Game Start</Heading>

        <Flex direction="column" gap="2">
            <Flex direction="row" gap="2">
            <Button onClick={() => executeMoveCall("small")}>
                small
            </Button>
            <Button onClick={() => executeMoveCall("large")}>
                large
            </Button>
            </Flex>
        </Flex>
        </>
    );
}
```

### Response

> Example transaction：https://suiexplorer.com/txblock/2hg7QQvNh7DnmmMaanAZ7hRYCAUimPJTN3bJRnpo29ej?network=testnet

Reviewer: ```suggestion
When the subscribed to event fires, the example displays the following JSON representation of the event.

{
  "id": {
    "txDigest": "2hg7QQvNh7DnmmMaanAZ7hRYCAUimPJTN3bJRnpo29ej",
    "eventSeq": "0"
  },
  "packageId": "<PACKAGE-ID>",
  "transactionModule": "<MODULE>",
  "sender": "<SENDER-ADDRESS>",
  "type": "<PACKAGE-ID>::<MODULE>::Result",
  "parsedJson": {
    "msg": "You Win :)"
  },
  "bcs": "3ZrJhsA3GhA8gKr",
  "timestampMs": "1707120095029"
}
```
Can't use a link to testnet as data is not guaranteed to persist.

> I can get the value of parsedJson in the example below, and its final value is {"msg":"You Win :)"} .

Reviewer: ```suggestion
As the returned JSON shows, the value of `parsedJson` is `{"msg":"You Win :)"}` .
```

Author: Thanks @ronny-mysten ，you're probably right.
